### PR TITLE
Mepts 263

### PR DESCRIPTION
**Pull request by ningosi**
Added E and F section of the Resumo mensal report for https://jembiprojects.jira.com/browse/MEPTS-263, there is need to have patient data simulation that will return the results for those indicators

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java [hunk 1]
```diff
@@ -479,4 +485,384 @@ private DateObsCohortDefinition getLastArvPickupDateCohort() {
...
+        "VLQ",
+        map(
+            getPatientsToExclude(
+                hivMetadata.getHivViralLoadQualitative(),
+                hivMetadata.getAdultoSeguimentoEncounterType(),
+                "CODED",
+                "NO",
+                "obs"),
+            "onOrAfter=${startDate},onOrBefore=${endDate},location=${location}"));
+    cd.setCompositionString("(C AND VL) AND (VLX OR VLQ)");
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: This composition seems wrong: should it be: `(C AND VL) AND NOT (VLX OR VLQ)`?

note the exclusion...

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java [hunk 2]
```diff
@@ -479,4 +485,384 @@ private DateObsCohortDefinition getLastArvPickupDateCohort() {
...
+        "Ex2",
+        map(
+            getPatientsToExclude(
+                hivMetadata.getHivViralLoadQualitative(),
+                hivMetadata.getAdultoSeguimentoEncounterType(),
+                "CODED",
+                "NO",
+                "obs"),
+            "onOrAfter=${startDate},onOrBefore=${endDate},location=${location}"));
+    cd.setCompositionString("(C AND (SUPP OR QUAL)) AND (Ex1 OR Ex2)");
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: Same comment here about the exclusion, shouldn't we be excluding Ex1 & Ex2?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/resumo/ExcludeCriteriaForECalculation.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    int day = calendar.get(Calendar.DAY_OF_MONTH);
+    int month = calendar.get(Calendar.MONTH);
+    Concept concept = (Concept) parameterValues.get("concept");
+    EncounterType encounterType = (EncounterType) parameterValues.get("encounterType");
+    String type = (String) parameterValues.get("type");
+    String limit = (String) parameterValues.get("limit");
+    String option = (String) parameterValues.get("option");
+
+    if (!(day == 21 && month == 11)) {
+      requiredDate = EptsCalculationUtils.addMonths(startDate, -12);
```
[the remark this answers is not in the document]
Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/resumo/ExcludeCriteriaForECalculation.java]: Ok. Looking into rewriting those

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java [hunk 4]
```diff
@@ -479,4 +485,384 @@ private DateObsCohortDefinition getLastArvPickupDateCohort() {
...
+        "Ex",
+        map(
+            getPatientsToExclude(
+                hivMetadata.getApplicationForLaboratoryResearch(),
+                hivMetadata.getAdultoSeguimentoEncounterType(),
+                "ENC",
+                "NO",
+                "encounter"),
+            "onOrAfter=${startDate},onOrBefore=${endDate},location=${location}"));
+    cd.setCompositionString("F1 AND Ex");
```
Comment (nthfloor) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: Same comment here about the exclusion...seems it should be `...NOT Ex`

**Pull request merged**